tiny_dnn/network.h

```diff
@@ -486,33 +486,34 @@ class network {
       if (current->weights().size() < 2) {
         continue;
       }
-      vec_t &w     = *current->weights()[0];
-      vec_t &b     = *current->weights()[1];
-      tensor_t &dw = (*current->weights_grads()[0]);
-      tensor_t &db = (*current->weights_grads()[1]);
+      std::vector<vec_t *> weights     = current->weights();
+      std::vector<tensor_t *> grads = current->weights_grads();
 
-      if (w.empty()) continue;
+      if (weights.empty() || (*weights[0]).empty()) continue;
```

Comment:
Should I change it then? I can do it in just a moment but I think it is correct both ways.